### PR TITLE
Added alternate stats displaying the count for each rating

### PR DESCRIPTION
**Pull request by SamZhang02**

### CODE REVIEW

File: client/src/components/CourseInfo.tsx [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
+import { countRatings, classNames } from '../lib/utils';
+import _ from 'lodash';
+
+export type map = {
+  1: number;
+  2: number;
+  3: number;
+  4: number;
+  5: number;
+};
```
Comment (JeffDotPng) [inline client/src/components/CourseInfo.tsx:17]: Please pick a better name for this type (also why not just use an array or tuple)?

File: client/src/components/RatingHistogram.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      <div className='flex-auto'>
+        <Bar num={num} total={total} />
+      </div>
+      <p className='w-3 flex-none dark:text-gray-200'>{num}</p>
+    </div>
+  );
+};
+
+export const RatingHistogram = ({ ratings }: RatingHistogramProps) => {
+  const total = ratings[1] + ratings[2] + ratings[3] + ratings[4] + ratings[5];
```
Comment (JeffDotPng) [inline client/src/components/RatingHistogram.tsx:37]: You can use lodash's sum function here: `_.sum(Object.values(ratings))`

File: client/src/components/RatingHistogram.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <div className='flex w-full flex-col space-y-4'>
+      <IndividualRating rating={5} num={ratings[5]} total={total} />
+      <IndividualRating rating={4} num={ratings[4]} total={total} />
+      <IndividualRating rating={3} num={ratings[3]} total={total} />
+      <IndividualRating rating={2} num={ratings[2]} total={total} />
+      <IndividualRating rating={1} num={ratings[1]} total={total} />
+    </div>
```
Comment (JeffDotPng) [inline client/src/components/RatingHistogram.tsx:46]: Just map over `[5,4,3,2,1]`

File: client/src/components/RatingInfo.tsx [hunk 4]
```diff
@@ -1,27 +1,55 @@
...
 }: {
   title: string;
-  rating: number;
+  chartType: 'pie' | 'histogram';
   numReviews: number;
+  ratings: map;
   content?: string;
 }) => {
+  const averageRating =
+    (1 * ratings[1] +
```
Comment (JeffDotPng) [inline client/src/components/RatingInfo.tsx:19]: Can use map + sum here

File: client/src/components/RatingInfo.tsx [hunk 5]
```diff
@@ -1,27 +1,55 @@
...
+      2 * ratings[2] +
+      3 * ratings[3] +
+      4 * ratings[4] +
+      5 * ratings[5]) /
+    numReviews;
+
+  const chart = (type: 'pie' | 'histogram') => {
```
Comment (JeffDotPng) [inline client/src/components/RatingInfo.tsx:26]: This doesn't need to be a function, just store the element in a variable

File: client/src/components/RatingHistogram.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
+type RatingHistogramProps = { ratings: any };
```
Comment (JeffDotPng) [inline client/src/components/RatingHistogram.tsx:1]: why?

File: client/src/components/Toggle.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        <span className='sr-only'>Use setting</span>
+        <span
+          aria-hidden='true'
+          className={`${enabled ? 'translate-x-9' : 'translate-x-0'}
+            pointer-events-none inline-block h-[34px] w-[34px] transform rounded-full bg-white shadow-lg ring-0 transition duration-200 ease-in-out`}
+        />
+      </Switch>
+    </div>
+  );
+};
```
Comment (JeffDotPng) [inline client/src/components/Toggle.tsx:28]: This whole component is not used

File: client/src/lib/utils.ts [hunk 8]
```diff
@@ -91,3 +92,30 @@ export const sortSchedulesByBlocks = (schedules: Schedule[]) => {
...
+  reviews: Review[]
+) => {
+  type validRating = 1 | 2 | 3 | 4 | 5;
+  const ratings = {
+    1: 0,
+    2: 0,
+    3: 0,
+    4: 0,
+    5: 0,
+  };
```
Comment (JeffDotPng) [inline client/src/lib/utils.ts:107]: We should honestly just use an array instead, it's just simpler

File: client/src/pages/Instructor.tsx [hunk 9]
```diff
@@ -16,6 +18,8 @@ export const Instructor = () => {
   const [reviews, setReviews] = useState<Review[]>([]);
   const [instructor, setInstructor] = useState<InstructorType | null>(null);
   const [showAllReviews, setShowAllReviews] = useState(false);
+  const [chartType, setChartType] = useState<'pie' | 'histogram'>('pie');
+  console.log(chartType);
```
Comment (JeffDotPng) [inline client/src/pages/Instructor.tsx:22]: Should remove this console.log

File: client/src/components/RatingHistogram.tsx (hunk 6 above)
Comment (SamZhang02) [reply client/src/components/RatingHistogram.tsx:1]: oops i was trying something out XD

File: client/src/components/CourseInfo.tsx (hunk 1 above)
Comment (SamZhang02) [reply client/src/components/CourseInfo.tsx:17]: i just thought that it had a bit better readability, since arrays index from 0

**Comment by terror** [comment]
Closing in favour of #366

**Pull request closed without merging**